src/reanimated2/frameCallback/FrameCallbackRegistryUI.ts

```diff
@@ -1,34 +1,71 @@
 import { runOnUI } from '../core';
 
 export default interface FrameCallbackRegistryUI {
-  frameCallbackRegistry: Map<number, () => void>;
+  frameCallbackRegistry: Map<number, (frameTimings: FrameTimings) => void>;
+  frameCallbackStartTime: Map<number, number>;
   frameCallbackActive: Set<number>;
   isFrameCallbackRunning: boolean;
+  lastFrameTimestamp: number;
   runCallbacks: () => void;
-  registerFrameCallback: (callback: () => void, callbackId: number) => void;
+  registerFrameCallback: (
+    callback: (frameTimings: FrameTimings) => void,
+    callbackId: number
+  ) => void;
   unregisterFrameCallback: (frameCallbackId: number) => void;
   manageStateFrameCallback: (frameCallbackId: number, state: boolean) => void;
 }
 
+export type FrameTimings = {
+  timestamp: number;
+  timeSinceLastFrame: number | null;
+  elapsedTime: number;
+};
+
 export const prepareUIRegistry = runOnUI(() => {
   'worklet';
 
   const frameCallbackRegistry: FrameCallbackRegistryUI = {
-    frameCallbackRegistry: new Map<number, () => void>(),
+    frameCallbackRegistry: new Map<
+      number,
+      (frameTimings: FrameTimings) => void
+    >(),
+    frameCallbackStartTime: new Map<number, number>(),
     frameCallbackActive: new Set<number>(),
     isFrameCallbackRunning: false,
+    lastFrameTimestamp: 0,
```

Comment:
Why not `undefined` or `null`? This way we can easily detect first frame. Yeah, I know, ain't nobody using Reanimated in 1970.

Reply:
`undefned` sounds good